packages/sync/src/firehose/index.ts

```diff
@@ -0,0 +1,342 @@
+    }
+  })
+  const verifiedCids: Record<string, CID | null> = {}
+  try {
+    const results = await verifyProofs(evt.blocks, claims, did, key)
+    results.verified.forEach((op) => {
+      const path = formatDataKey(op.collection, op.rkey)
+      verifiedCids[path] = op.cid
+    })
+  } catch (err) {
+    if (err instanceof RepoVerificationError) {
+      return parseCommitAuthenticated(idResolver, evt, true)
+    }
+    throw err
+  }
+  const verifiedOps: RepoOp[] = evt.ops.filter((op) => {
+    if (op.action === 'delete') {
+      return verifiedCids[op.path] === null
+    } else {
+      return op.cid !== null && op.cid.equals(verifiedCids[op.path])
+    }
+  })
+  return formatCommitOps(evt, verifiedOps)
+}
+
+export const parseCommitUnauthenticated = async (
+  evt: Commit,
+): Promise<CommitEvt[]> => {
+  return formatCommitOps(evt, evt.ops)
+}
+
+const formatCommitOps = async (evt: Commit, ops: RepoOp[]) => {
+  const car = await readCar(evt.blocks)
+
+  const evts: CommitEvt[] = []
+
+  for (const op of ops) {
+    const uri = new AtUri(`at://${evt.repo}/${op.path}`)
+
+    const meta: CommitMeta = {
+      seq: evt.seq,
+      time: evt.time,
+      commit: evt.commit,
+      blocks: car.blocks,
+      rev: evt.rev,
+      uri,
+      did: uri.host,
+      collection: uri.collection,
+      rkey: uri.rkey,
+    }
+
+    if (op.action === 'create' || op.action === 'update') {
+      if (!op.cid) continue
+      const recordBytes = car.blocks.get(op.cid)
+      if (!recordBytes) continue
+      const record = cborToLexRecord(recordBytes)
+      evts.push({
+        ...meta,
+        event: op.action as 'create' | 'update',
+        cid: op.cid,
+        record,
+      })
+    }
+
+    if (op.action === 'delete') {
+      evts.push({
+        ...meta,
+        event: 'delete',
+      })
+    }
+  }
+
+  return evts
+}
+
+export const parseIdentity = async (
+  idResolver: IdResolver,
+  evt: Identity,
+  unauthenticated = false,
+): Promise<IdentityEvt | null> => {
+  const res = await idResolver.did.resolve(evt.did)
+  const handle =
+    res && !unauthenticated
+      ? await verifyHandle(idResolver, evt.did, res)
+      : undefined
+
+  return {
+    event: 'identity',
+    seq: evt.seq,
+    time: evt.time,
+    did: evt.did,
+    handle,
+    didDocument: res,
+  }
+}
+
+const verifyHandle = async (
+  idResolver: IdResolver,
+  did: string,
+  didDoc: DidDocument,
+): Promise<string | undefined> => {
+  const { handle } = parseToAtprotoDocument(didDoc)
+  if (!handle) {
+    return undefined
+  }
+  const res = await idResolver.handle.resolve(handle)
+  return res === did ? handle : undefined
+}
+
+export const parseAccount = (evt: Account): AccountEvt | undefined => {
+  if (evt.status && !isValidStatus(evt.status)) return
+  return {
+    event: 'account',
+    seq: evt.seq,
+    time: evt.time,
+    did: evt.did,
+    active: evt.active,
+    status: evt.status as AccountStatus,
```

Comment:
Based on the check above, could `evt.status` also be missing (in addition to `AccountStatus`)?

Reply:
not sure i follow?

Reply:
`evt.status` is being casted to `AccountStatus`, but from looking at some of the lines of code above I believe it also may be missing.

Reply:
We have passed the check:
```ts
if (evt.status && !isValidStatus(evt.status)) return
```

So that means `evt.status` is either falsy or it's a valid `AccountStatus`.

Reply:
ahhh got it got it 👍